### ISSUE TITLE
Fixing "New Cards Added" statistic has broken Total/Average Added Cards reporting for pre-made template/skeleton decks

### ISSUE THREAD

**Issue by quarridors** [ankidroid/Anki-Android#7160]
###### Reproduction Steps

1. Unsuspend cards in a pre-made deck
2. Learn those cards
3. Switch to 'Statistics'
3. Observe the figures under 'Added', both 'Total' and 'Average' in the 'Overview' tab

###### Expected Result

Before version 2.13.0 (release notes include 'Fix "New Cards Added" Statistic'), previously suspended newly learned cards were counted in the total added cards and average added cards per day statistics.

This was desired behaviour in my case as I am using a pre-made skeleton deck of partially filled cards that accompanies books on learning Chinese characters (essentially a 'template deck' - but not using the template functionality).

The system in the books teaches the reader to write their own mnemonic stories to memorise the characters, and encourages the use of personalised flashcards for learning. As everyone's flashcards for the book will have the same characters and keywords, there are pre-made skeleton Anki decks available to speed up this process. Each card in those decks has the Chinese character and the keyword from the books (plus other enhancements like stroke order animations etc, depending on the deck). Each card should start suspended because the main mnemonic story field is blank, and cards are only to be added to rotation once the user has learned a mnemonic for them.

As one works through the course in the books, the process of 'adding' new cards to your deck involves finding the appropriate suspended partial/skeleton card, writing a mnemonic story into the blank field, unsuspending the card and beginning the Anki learning process.

The previous behaviour of the 'Added' statistics counted each of those newly unsuspended and learned cards as 'newly added' and gave me a total number of those cards added during the reporting period, and an average for all the days in the period. For example, the previous version of AnkiDroid told me I had exceeded 1800 total cards added over the deck life.

It was effectively reporting on the number of cards that were highlighted as blue in the due count at the time they were learned.

My workflow with AnkiDroid has been aiming to consistently maintain an average of 15 cards per day or more added/newly learned in the 30 day statistics overview. The previous version told me that my average for this month was 13 cards a day so I was planning to use this weekend to catch up.

###### Actual Result

Since the 2.13.0 update (which includes 'Fix "New Cards Added" Statistic' in the release notes), none of the previously counted, suspended partially-filled/skeleton cards that I've been adding to rotation as I learn mnemonics are being counted as 'new cards added' in the statistics. I can add 20 new cards into rotation in a day and be told by this statistic that none were added. The Learned figure under the 'Today' summary doesn't match the 'Added' total.

If I add new cards from scratch, these are counted, but this is something that happens extremely rarely with the way I use AnkiDroid. Previously these also counted towards to total and average but were not the only type of cards reported in that total.

When AnkiDroid updated versions, the statistic changed overnight from nearly 400 cards added in the current 30 day period to giving a total of 13 added cards (the number added from scratch).

As the release notes included 'Fix "New Cards Added" Statistic', I assume that this is an intentional behaviour change, rather than a bug. I can see the logic in only wanting the cards that have been added from scratch to be included in this statistic. But I also can find no way to restore the previous behaviour through changing the settings for statistics and reports in AnkiDroid.

If this isn't considered a bug, but an intentional functionality change, I would like the option to include newly learned cards (that show up as the blue 'Learn' colour in Review Count/Time charts) in the total and average calculated as 'Added' during this period. Alternatively, I'd like this figure to at least be included alongside 'added' with a similar cumulative total and daily average.

With this recent functionality change, I now only have the 'Learn' figure as given at the top in the 'Today' section to work from, which is only given for the current day. There's no average presented. I can approximate the previous version's average new cards added figure by estimating the number from the blue line in Review Count chart and then manually divide this by 30, but this is not exactly convenient for the end user, or as exact.

As my workflow previously involved keeping the daily average of newly learned cards in the last 30 days at a particular level, this behaviour change to statistics calculation has negatively impacted my use of AnkiDroid.

###### Debug info
AnkiDroid Version = 2.13.1

Android Version = 9

ACRA UUID = 9cae9198-9773-489b-b08d-940acf072fb2

Scheduler = std

###### Research
*Enter an [x] character to confirm the points below:*

- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by quarridors** [comment]
I've just made an additional discovery that suggests this may actually be an unintended bug. I just noticed that while the year and month reports are giving the number of cards added from scratched as the reported total, the 'Deck Life' report is giving the number of cards total in the entire deck, even when the deck's life is shorter than 1 year so should match the year report.

Worse still, the average per day figures on the Deck Life reports don't seem correct, one of my decks is 251 days old, claims 3238 cards were added but gives the average as 1.7 cards per day. I make 3238/251 to work out as 12.9 cards per day. 

Same goes for the All Decks report, last month was 14 cards added, 0.5 per day, last year was 14 cards added, 0 per day, Deck Life is 31922 cards, 16.9 cards per day, 284 days for the reported period. I make 31922/284 as 112.4 cards per day.

**Comment by david-allison-1** [comment]
@quarridors Could you provide a sample where this differs from the stats shown in Anki Desktop. I'll get to it.

**Comment by quarridors** [comment]
> @quarridors Could you provide a sample where this differs from the stats shown in Anki Desktop. I'll get to it.

@david-allison-1 I'm not sure that the Mac version has an equivalent of this figure, I couldn't find it when I started using Anki and quickly switched to the Android version due to it giving totals of and averages of the number of cards I'd newly learned in a month.

I'm downloading a new version of the Mac client now to see if I can find it.

**Comment by quarridors** [comment]
@david-allison-1 Okay the new Mac version has installed, reporting looks completely different from the last time I used it.

There is an 'Added' report right at the very bottom, this gives all the cards I've manually added myself.

That seems to have a different but related bug. The 'All' option under Added, which is presumably the equivalent of Deck Life, says that 3,238 cards were added for the 251 day old deck and that it works out as 34 cards per day, which is wrong in a different way. The chart however goes back to day -1,860 and shows no cards actually being added anywhere in the chart except in the last 39 to 20 days, when a total of 14 were added.

The 1 month figure on Mac version seems to use a 31 day period rather than AnkiDroid's 30 days. It says 14 cards were added in that total but gives the average per day as 0 cards, when AnkiDroid gave 0.5 per day.

So that's the desktop version clearly differing from AnkiDroid ...but both seem to have related bugs that are breaking their 'deck life' or 'all' reporting period, albeit in different ways, which suggests that the bug is probably located in some backend code that both share?

Should I report this somewhere else? Any idea what the correct behaviour of the 'Added' report is meant to be? What AnkiDroid was previously doing, or what it does now (deck life counting bug aside)?
